meshmode/discretization/poly_element.py

```diff
@@ -525,12 +525,29 @@ def basis_obj(self):
 
     @memoize_method
     def quadrature_rule(self):
-        basis = self._basis
-        nodes = self._nodes
-        mass_matrix = mp.mass_matrix(basis, nodes)
-        weights = np.dot(mass_matrix,
-                         np.ones(len(basis.functions)))
-        return mp.Quadrature(nodes, weights, exact_to=self.order)
+        from modepy.tools import reshape_array_for_tensor_product_space as fold
+
+        quads = []
+
+        if self.dim != 1:
+            nodes_tp = fold(self.space, self._nodes)
+        else:
+            nodes_tp = self._nodes
+
+        for idim, zipped in enumerate(zip(nodes_tp, self._basis.bases)):
```

Comment:
```suggestion
        for idim, (nodes, basis) in enumerate(zip(nodes_tp, self._basis.bases)):
```
should unpack it correctly, so there's no need for the next line.